### PR TITLE
CUDA RDC: add support for MODULE library

### PR DESCRIPTION
**Pull request by pcanal**
And a few cleanups (See STREQUAL usage)

### CODE REVIEW

**Review by sethrj — APPROVED** [review]
Looks good to me, I guess it was quite a simple change after all. You said that replacing this with the standalone https://github.com/pcanal/CudaRDCExercise will happen in a separate commit?

File: cmake/CeleritasLibrary.cmake [hunk 1]
```diff
@@ -727,9 +733,9 @@ function(celeritas_target_link_libraries target)
     celeritas_cuda_gather_dependencies(_flat_target_link_libraries ${_target_middle})
     foreach(_lib ${_flat_target_link_libraries})
       get_target_property(_lib_target_type ${_lib} TYPE)
-      if(NOT "x${_lib_target_type}" STREQUAL "xINTERFACE_LIBRARY")
+      if(NOT _lib_target_type STREQUAL "INTERFACE_LIBRARY")
         celeritas_check_cuda_runtime(_lib_runtime_setting ${_lib})
-        if (NOT ${_need_to_use_shared_runtime} AND ${_lib_runtime_setting} STREQUAL "Shared")
+        if (NOT ${_need_to_use_shared_runtime} AND _lib_runtime_setting STREQUAL "Shared")
```
Comment (sethrj) [inline cmake/CeleritasLibrary.cmake]:
```suggestion
        if (NOT _need_to_use_shared_runtime AND _lib_runtime_setting STREQUAL "Shared")
```

File: cmake/CeleritasLibrary.cmake [hunk 2]
```diff
@@ -273,14 +273,19 @@ function(celeritas_rdc_add_library target)
...
-  if((NOT BUILD_SHARED_LIBS AND NOT _ADDLIB_PARSE_SHARED)
+  if((NOT BUILD_SHARED_LIBS AND NOT _ADDLIB_PARSE_SHARED AND NOT _ADDLIB_PARSE_MODULE)
       OR _ADDLIB_PARSE_STATIC)
     set(_lib_requested_type "STATIC")
     set(_cudaruntime_requested_type "Static")
     set(_staticsuf "")
   endif()
-  if(_ADDLIB_PARSE_MODULE)
-    message(FATAL_ERROR "celeritas_rdc_add_library does not support MODULE library containing device code")
+  if (_ADDLIB_PARSE_MODULE)
```
Comment (sethrj) [inline cmake/CeleritasLibrary.cmake]:
```suggestion
  if(_ADDLIB_PARSE_MODULE)
```

File: cmake/CeleritasLibrary.cmake [hunk 3]
```diff
@@ -131,21 +131,21 @@ define_property(TARGET PROPERTY CELERITAS_CUDA_OBJECT_LIBRARY
...
-        "x${arg}" STREQUAL "xEXCLUDE_FROM_ALL" OR
-        "x${arg}" STREQUAL "xSTATIC" OR
-        "x${arg}" STREQUAL "xSHARED" OR
-        "x${arg}" STREQUAL "xMODULE"
+        arg STREQUAL "WIN32" OR
+        arg STREQUAL "MACOSX_BUNDLE" OR
+        arg STREQUAL "EXCLUDE_FROM_ALL" OR
+        arg STREQUAL "STATIC" OR
+        arg STREQUAL "SHARED" OR
+        arg STREQUAL "MODULE"
```
Comment (sethrj) [inline cmake/CeleritasLibrary.cmake:148]: Maybe simplify to `arg MATCHES "^[A-Z_]+$"`? All source files will have an extension.

File: cmake/CeleritasLibrary.cmake [hunk 4]
```diff
@@ -168,11 +168,11 @@ function(celeritas_sources_contains_cuda var)
   set(_result)
   foreach(_source ${ARGN})
     get_source_file_property(_iscudafile "${_source}" LANGUAGE)
-    if ("x${_iscudafile}" STREQUAL "xCUDA")
+    if (_iscudafile STREQUAL "CUDA")
```
Comment (drbenmorgan) [inline cmake/CeleritasLibrary.cmake]:
```suggestion
    if(_iscudafile STREQUAL "CUDA")
```

**Review by drbenmorgan — APPROVED** [review]
All looks good to me! Have tested on the FullSimLight module library that exposed the problem and confirmed the changes here resolve the issue, so good to go on that side of things.

**Review by sethrj — CHANGES_REQUESTED** [review]
I noticed a couple of places that could use fixing/improving

**Comment by pcanal** [comment]
> You said that replacing this with the standalone [pcanal/CudaRDCExercise](https://github.com/pcanal/CudaRDCExercise) will happen in a separate commit?

A separate PR as it is semantically unrelated.

File: cmake/CeleritasLibrary.cmake (hunk 3 above)
Comment (pcanal) [reply cmake/CeleritasLibrary.cmake:148]: Yes, I could do that .. note that this code is literally inspired by Kitware's code in FindCUDA.mk: Kitware/CMake@b8b53db/Modules/FindCUDA.cmake#L1327

**Comment by sethrj** [comment]
So is the replacement of "x" which made this harder to review 😬

File: cmake/CeleritasLibrary.cmake (hunk 3 above)
Comment (sethrj) [reply cmake/CeleritasLibrary.cmake:148]: The FindCUDA is *very* old but you have a point... I'm just concerned about more library arguments being added by kitware.

**Review by sethrj — APPROVED** [review]

**Pull request merged**